lib/cuetip/worker.rb

```diff
@@ -34,7 +35,13 @@ def run
     def run_once
       set_status('polling')
       run_callbacks :poll do
-        queued_job = silence { Cuetip::Models::QueuedJob.find_and_lock }
+        queued_job = silence do
+          if @queues.any?
+            Cuetip::Models::QueuedJob.from_queues(@queues).find_and_lock
+          else
+            Cuetip::Models::QueuedJob.find_and_lock
+          end
+        end
```

Comment:
```suggestion
          if @queues.any?
            scope = Cuetip::Models::QueuedJob.from_queues(@queues)
          else
            scope = Cuetip::Models::QueuedJob
          end
          scope.find_and_lock
        end
```

What about something like this?